### PR TITLE
Fix IPv6 remote addresses from X_FORWARDED_FOR headers before validating

### PR DESCRIPTION
**Pull request by nickvergessen**
Signed-off-by: Joas Schilling <coding@schilljs.com>

### CODE REVIEW

**Comment by ChristophWurst** [comment]
something tells me that this is not 100% correct but I do not recall the discussion.

What is the purpose of this?

**Comment by nickvergessen** [comment]
Well the header is apparently set to `[::1]` by proxies, but that doesn't validate:

```
<?php
$IP = '[2a02:120b:2c00:6670:ece9:7fff:4339:b972]';
var_dump(filter_var($IP, FILTER_VALIDATE_IP)); // false
$IP = trim($IP, '[]');
var_dump(filter_var($IP, FILTER_VALIDATE_IP)); // '2a02:120b:2c00:6670:ece9:7fff:4339:b972'
```

**Review by ChristophWurst — APPROVED** [review]

File: lib/private/AppFramework/Http/Request.php [hunk 1]
```diff
@@ -653,6 +653,12 @@ public function getRemoteAddress(): string {
 				if (isset($this->server[$header])) {
 					foreach (explode(',', $this->server[$header]) as $IP) {
 						$IP = trim($IP);
+
+						// remove brackets from IPv6 addresses
+						if (strpos($IP, '[') === 0 && substr($IP, -1) === ']') {
```
Comment (kesselb) [inline lib/private/AppFramework/Http/Request.php:658]: trim($ip, '[]'); should also work.

Comment (nickvergessen) [reply lib/private/AppFramework/Http/Request.php:658]: Yeah, but we use this code in other places so unity is prefered.

**Review by kesselb — APPROVED** [review]

**Pull request merged**